File: exercises/beecrowd/1049/1049.go
// https://judge.beecrowd.com/pt/problems/view/1049
package main

import (
	"fmt"
)

func main() {
	var category, class, diet string

	fmt.Scanln(&category)
	fmt.Scanln(&class)
	fmt.Scanln(&diet)

	animalMap := map[string]map[string]map[string]string{
		"vertebrado": {
			"ave": {
				"carnivoro": "aguia",
				"onivoro":   "pomba",
			},
			"mamifero": {
				"onivoro":   "homem",
				"herbivoro": "vaca",
			},
		},
		"invertebrado": {
			"inseto": {
				"hematofago": "pulga",
				"herbivoro":  "lagarta",
			},
			"anelideo": {
				"hematofago": "sanguessuga",
				"onivoro":    "minhoca",
			},
		},
	}

	if animal, ok := animalMap[category][class][diet]; ok {
		fmt.Println(animal)
	}
}

File: exercises/beecrowd/1049/1049.ts
// https://judge.beecrowd.com/pt/problems/view/1049
import * as fs from "fs";

function main(): void {
	const input = fs.readFileSync("/dev/stdin", "utf8");
	const lines = input.split(/\r?\n/);

	const category = lines.shift();
    const classType = lines.shift();
    const diet = lines.shift();

	const animalMap: Record<string, Record<string, Record<string, string>>> = {
		vertebrado: {
		  ave: {
			carnivoro: "aguia",
			onivoro: "pomba",
		  },
		  mamifero: {
			onivoro: "homem",
			herbivoro: "vaca",
		  },
		},
		invertebrado: {
		  inseto: {
			hematofago: "pulga",
			herbivoro: "lagarta",
		  },
		  anelideo: {
			hematofago: "sanguessuga",
			onivoro: "minhoca",
		  },
		},
	};
	  
	console.log(animalMap?.[category]?.[classType]?.[diet] || "");
}

main();
